src/formatDate.js

```diff
@@ -8,7 +8,60 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
-}
+  const fromSeparator = fromFormat[fromFormat.length - 1];
+  const toSeparator = toFormat[toFormat.length - 1];
+
+  const dateParts = date.split(fromSeparator);
+  const newDate = [];
+  const newDateObj = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (
+      fromFormat[i] === 'YYYY' ||
+      fromFormat[i] === 'YY' ||
+      fromFormat[i] === 'MM' ||
+      fromFormat[i] === 'DD'
+    ) {
+      newDateObj[fromFormat[i]] = dateParts[i];
+    }
+  }
+
+  const centuryThreshold = 30;
+  const pastCentury = '19';
+  const actualCentury = '20';
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    newDateObj['YY'] = newDateObj['YYYY'].slice(-2);
+  }
```

Comment:
The conversion from 'YYYY' to 'YY' is correct, but it's performed inside a loop that is not necessary since the condition is not dependent on a loop variable. This could be simplified by taking it out of any loop.